### PR TITLE
Use array unpacking to replace `array_merge`

### PR DESCRIPTION
**Pull request by distantnative**
## This PR …

I know you will laugh at me and slightly hate me... but I had this already ready for v5 😆 Since it looks like we're dropping PHP 8.0 with v4 already, this is potentially ripe for review. 

But given the vast of changes, even if little, I'd also be 100% fine if we put this off for 4.1 or so.

### CODE REVIEW

**Comment by bastianallgeier** [comment]
Yep, looking at it again, it feels like we shouldn't do it at this point. I think it would be good to keep it on a lab branch, but close the PR. We are already suffering from too many unclosed PRs again.

**Comment by afbora** [comment]
I've reviewed half of the PR. I'll finish the rest today. The changes are simple even though there are a lot of changes. `array_merge to [...]`. I'm not against including it in the beta process. I also loved the new unpacking feature 💛

File: src/Panel/Field.php [hunk 1]
```diff
@@ -39,11 +39,14 @@ public static function dialog(
...
 
 		foreach ($field->dialogs() as $dialogId => $dialog) {
-			$routes = array_merge($routes, Dialog::routes(
-				id: $dialogId,
-				areaId: 'site',
-				options: $dialog
-			));
+			$routes = [
+				...$routes,
+				Dialog::routes(
```
Comment (afbora) [inline src/Panel/Field.php]:
```suggestion
				...Dialog::routes(
```

**Comment by distantnative** [comment]
@afbora thank you - I suggest you add your review comments, I fix them and we then close the PR and I reopen it for 4.x

**Comment by afbora** [comment]
I found only one problem while reviewing. I have no other comments.

**Comment by distantnative** [comment]
Will open this again for 4.x

**Pull request closed without merging**